File: docs/competitors/lab1/php/CircleSquare.php
<?php
declare(strict_types=1);

define('INPUT_MESSAGE', "Введите радиус круга: ");
define('RESULT_MESSAGE', "Площадь вашего круга равна: %s\n");
define('ERROR_MESSAGE', "Пожалуйста, введите корректное положительное число для радиуса.\n");
define('MIN_RADIUS', 0);

echo INPUT_MESSAGE;
$radius = trim(fgets(STDIN));

if (is_numeric($radius) && $radius > MIN_RADIUS) 
{
    $area = pi() * $radius * $radius;
    printf(RESULT_MESSAGE, $area);
} 
else 
{
    echo ERROR_MESSAGE;
}
?>

File: docs/competitors/lab1/go/CircleSquare/CircleSquare.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strconv"
	"strings"
)

const (
	inputMessage      = "Введите радиус круга: "
	resultMessage     = "Площадь вашего круга равна: %f\n"
	errorMessage      = "Пожалуйста, введите корректное положительное число для радиуса.\n"
	
	stopReadingSymbol = '\n'
	minRadius         = 0
	bitPrecision      = 64
)

func main() {
	inputStream := bufio.NewReader(os.Stdin)

	fmt.Printf(inputMessage)
	input, _ := inputStream.ReadString(stopReadingSymbol)
	inputRadius := strings.TrimSpace(input)

	radius, err := strconv.ParseFloat(inputRadius, bitPrecision)
	if err != nil || radius <= minRadius {
		fmt.Printf(errorMessage)
		return
	}

	area := math.Pi * math.Pow(radius, 2)
	fmt.Printf(resultMessage, area)
}
